modules/tensor_mechanics/include/auxkernels/RankTwoScalarAux.h

```diff
@@ -29,12 +29,18 @@ class RankTwoScalarAux : public AuxKernel
 
 protected:
   virtual Real computeValue();
-  Real calcEigenValues();
 
   const MaterialProperty<RankTwoTensor> & _tensor;
   MooseEnum _scalar_type;
 
+  /// whether or not selected_qp has been set
+  const bool _has_selected_qp;
+
+  /// The std::vector will be evaluated at this quadpoint only
+  const unsigned int _selected_qp;
+
 private:
```

Comment:
There are uses for all three access qualifiers (public, protected, and private). We try to encourage code reuse through inheritance. Protected sufficiently prevents access from the outside but not from derived classes. Sometimes an internal state has to be protected even from derived classes, or an internal interface may change in the future and you chose not to expose it, then you need private. But those cases are rather rare and you should have a good reason.